apps/docs/components/hooks/usePagination.md

```diff
@@ -0,0 +1,334 @@
+---
+layout: AtomLayout
+hideBreadcrumbs: true
+description: An easy to use function for creating pagination.
+---
+
+# usePagination
+
+:::::: slot usage
+
+The `usePagination` allows you to add logic to your pagination UI. With this <!-- react -->hook<!-- end react --><!-- vue -->composable<!-- end vue -->  you can add methods for moving to the next/previous page or to the exact page by passing its number. To calculate how many pages are needed to be displayed the only required parameter is the total number of items. There can be also passed additional parameters to adjust pagination according to your needs.
+
+## Usage
+
+<!-- react -->
+
+```tsx
+import { usePagination, SfButton, SfIconChevronLeft, SfIconChevronRight } from '@storefront-ui/react';
+
+function CustomPagination() {
+   const { totalPages, pages, selectedPage, startPage, endPage, next, prev, onPageChange, maxVisiblePages } = usePagination(150);
+
+  return (
+    <nav className="flex justify-between border-t border-neutral-200" role="navigation" aria-label="pagination">
+      <SfButton
+        type="button"
+        aria-label="Go to previous page"
+        disabled={selectedPage <= 1}
+        variant="tertiary"
+        slotPrefix={<SfIconChevronLeft />}
+        onClick={() => prev()}
```

Comment:
```suggestion
        onClick={prev}
```
I think that should work, wdyt ? If that works can be changed everywhere, will not spam 3 more comments :)

Reply:
Unfortunately I got ts error that says: Type 'void' is not assignable to type 'MouseEventHandler<HTMLButtonElement> | undefined